### PR TITLE
Add Muse Glimmer 30B support

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- add strongly typed Muse Glimmer text/vision config extraction and registry entries
- implement the gated Muse text decoder, mixed sliding/full attention, dynamic packed vision tower, pixel shuffle adapter, projector, and multimodal embedding mixer
- export Muse Glimmer as decoder, vision encoder, and embedding ONNX models with text-only checkpoint support
- add tiny text/full-multimodal parity coverage plus real 30B BF16 L4/L5 golden references pinned to `97c77dff50b2797bcc558fa2d909761dbc575c59`
- enable CUDA Graph capture by default for autoregressive decoder sessions, keep one-shot multimodal sessions uncaptured, and retain an explicit per-decoder opt-out for incompatible models
- lower all Muse norms to fused ONNX RMSNormalization and residual SkipSimplifiedLayerNormalization operators

## Validation

- `3655 passed, 48 skipped` - full non-integration suite
- `149 passed` - ORT GenAI config and export regression suite
- tiny Hugging Face vs ONNX text and full vision -> embedding -> decoder numerical parity
- full-size 30B decoder, vision encoder, and embedding graph construction
- L4 real-weight BF16 CUDA prefill passed on NVIDIA H200
- L5 real-weight BF16 CUDA 30-token greedy generation passed on NVIDIA H200
- real 30B INT4 CUDA text and image generation passed with decoder CUDA Graph capture and shared KV-cache buffers enabled
- full 30B BF16 CUDA graph contains 209 RMSNormalization and 104 SkipSimplifiedLayerNormalization nodes, with no decomposed ReduceMean/Pow norm path

The golden case is skipped in CI because the checkpoint is about 60 GB and requires a large-memory GPU; its committed references and local L4/L5 path remain fully runnable.

### CODE REVIEW

File: src/mobius/_config_resolver.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ def _dict_to_pretrained_config(d: dict):
     works correctly.
     """
     import transformers
+    from huggingface_hub.errors import StrictDataclassClassValidationError
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/_config_resolver.py:126]: Fixed in 0bd9f5c. The exception type is now resolved with getattr and falls back to TypeError for older huggingface_hub releases. Added a regression test that removes the newer symbol.

File: src/mobius/models/muse_glimmer.py [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+            attention_mask=attention_mask,
+            dtype=self._dtype,
+        )
+        sliding_attention_bias = create_attention_bias(
+            op,
+            input_ids=hidden_states if input_ids is None else input_ids,
+            attention_mask=attention_mask,
+            sliding_window=self._sliding_window,
+            dtype=self._dtype,
+        )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/muse_glimmer.py:302]: Fixed in 0bd9f5c. The decoder path now uses rank-2 position_ids as the query-length reference for both full and sliding attention masks.

File: tests/synthetic_parity_test.py [hunk 3]
```diff
@@ -502,6 +502,18 @@ def _create_hf_config(model_type: str, config_overrides: dict):
     if model_type in _HF_EXTRA_CONFIG:
         hf_kwargs.update(_HF_EXTRA_CONFIG[model_type])
 
+    if hf_model_type == "muse_glimmer_text":
+        hf_kwargs["head_dim"] = TINY_HEAD_DIM
+        hf_kwargs["hidden_activation"] = hf_kwargs.pop("hidden_act", "silu")
+        hf_kwargs["attention_bias"] = False
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/synthetic_parity_test.py]: Fixed in 0bd9f5c. An explicit hidden_activation now takes precedence; hidden_act is only used as the fallback alias.

File: tests/synthetic_parity_test.py [hunk 4]
```diff
@@ -665,11 +677,36 @@ def _create_hf_config(model_type: str, config_overrides: dict):
...
+                def forward(self, **kwargs):
+                    hidden_states = self.model(**kwargs).last_hidden_state
+                    logits = self.lm_head(hidden_states)
+                    logits = logits * hf_config.output_multiplier
+                    cap = hf_config.final_logit_softcapping
+                    return type(
+                        "MuseGlimmerOutput",
+                        (),
+                        {"logits": cap * torch.tanh(logits / cap)},
+                    )()
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/synthetic_parity_test.py]: Fixed in 0bd9f5c. The synthetic reference now returns unsoftcapped logits when the cap is zero or None, matching the ONNX implementation.

**Comment by justinchuby** [comment]
Added dedicated text-only L5 coverage in `0f34196`. The case forces `muse_glimmer_text`, loads the language backbone from the composite 30B checkpoint, commits a 20-token BF16 golden sequence, and passed real-weight ONNX generation on H200 (`1 passed in 713.32s`).

File: src/mobius/models/muse_glimmer.py [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+        image_features: ir.Value,
+    ):
+        text_embeddings = self.embed_norm(
+            op,
+            self.embed_tokens(op, input_ids),
+        )
+        feature_mask = op.Or(
+            op.Equal(input_ids, self._image_token_id),
+            op.Equal(input_ids, self._video_token_id),
+        )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/muse_glimmer.py]: Fixed in 5ee7332. The single tensor now has an explicit image-first, video-second packed contract, and each modality gets independent flattened cumulative indices. Added coverage with video placeholders before images across two batch rows to verify the streams cannot swap.

**Comment by justinchuby** [comment]
Olive/ORT GenAI validation found one final runtime-export issue: composite Muse builds expose `muse_glimmer_text` during processor generation, so they were missing the packed `PatchImage`/`image_grid_thw` pipeline. Commits `f225c65` and `5d87411` fix both parent and unwrapped config forms.

With that fix, the full 30B BF16→INT4 Olive recipe completed in 1,436 seconds (18 GB output), native CUDA text generation passed, and image generation correctly described a Chinatown archway scene through microsoft/onnxruntime-genai#2397.

**Comment by justinchuby** [comment]
Final H200 validation for `c076560` completed:

- Full BF16 -> INT4 Olive rebuild succeeded (736.8s export, 718.5s quantization).
- Corrected scale-free RMSNorm uses schema-valid 1-D scales; isolated CUDA output exactly matched the float32 reference.
- Final decoder: 2,458 nodes vs 3,913 baseline (-37.2%), 312 `RMSNormalization`, zero `ReduceMean`/`Pow`.
- ORT GenAI text and image E2E both passed with decoder CUDA Graph capture and shared KV buffer enabled.
- Same-runtime H200 benchmark: 61.76 tok/s median (61.74-61.82), up 11.29% from the 55.50 tok/s decomposed baseline.
- Final INT4 artifact and ONNX GenAI `inference_metadata.yaml` are published at https://huggingface.co/justinchuby/Muse-Glimmer-30B-ONNX-INT4-CUDA and all 13 uploaded package files were SHA-256 verified.
